frontend/src/utils.ts

```diff
@@ -0,0 +1,63 @@
+import axios from 'axios'
+import { useAppStore } from './stores/stores'
+import type { Project } from '@/constants'
+
+/** Checks if a given project ID is valid */
+export async function validProjectID(id: string | null) {
+  const storeApp = useAppStore()
+  let result = false
+  if (id == null) {
+    return result
+  }
+  try {
+    await axios.get<Project>(`${storeApp.constants.env.NAUTILUS_WEB_API}/projects/${id}`)
+    result = true
+  } catch (error: any) {
+    console.log(error)
+    storeApp.alertsError(error.message)
+    if (axios.isAxiosError(error) && error.response?.status == 404) {
+      result = false
+    }
+  }
+  return result
+}
+
+/** Checks if the User cookie is valid */
+export async function validateUser() {
+  const storeApp = useAppStore()
+  let result = true
+  try {
+    await axios.get<Project[]>(`${storeApp.constants.env.NAUTILUS_WEB_API}/projects`)
+    result = true
+  } catch (error: any) {
+    console.log(error)
+    storeApp.alertsError(error.message)
+    if (axios.isAxiosError(error) && error.response?.status == 401) {
+      result = false
+    }
+  }
+  return result
+}
+
+/** Clear all cookies */
+export function clearCookies() {
```

Comment:
> If we were to manipulate cookies client-side, I'd recommend using a library that would abstract this kind of thing.
But I think we shouldn't manipulate it here and we should clear the cookie server-side when validated_user fails.

I don't understand here How could I clear the cookie server-side when `validated_user` fails.

> This is important as we have no interest in manipulating the cookie here and we should actually make it httpOnly and Secure

What do `httpOnly` and `Secure` mean?

Reply:
> How could I clear the cookie server-side when `validated_user` fails.

You'd not read the cookie in JS but instead make a call to the API as you were doing previously. Like to retrieve list of projects or whatever. If it succeeds, your cookie is OK.

`validate_user` returns a `401` response in case it's not. This 401 should also use set_cookie to clear the cookie.

> What do `httpOnly` and `Secure` mean?

`httpOnly` means the cookie is only sent and received by the browser as HTTP headers and not exposed to manipulate via JS API. You can see it in the dev-console but can't read nor edit from JS.

Secure means it's only sent over HTTPS and localhost. Those are two options you can pass to `set_cookie`.